src/goto-programs/remove_function_pointers.cpp

```diff
@@ -359,6 +373,117 @@ void remove_function_pointerst::remove_function_pointer(
   remove_function_pointer(goto_program, function_id, target, functions);
 }
 
+
+/// For a call site, return a list of potential targets.
+/// param goto_program: the goto_program containing the call site
+/// param target: the actual callsite
+/// return: A list of potential function identifiers that can be
+///         the targets of this callsite.
+remove_function_pointerst::possible_fp_targetst
+remove_function_pointerst::list_potential_targets(
+  goto_programt &goto_program,
+  goto_programt::targett target)
+{
+  remove_function_pointerst::possible_fp_targetst targets;
+  bool found_functions;
+
+  const code_function_callt &code =
+    to_code_function_call(target->code);
+
+  const exprt &function = code.function();
+
+  // this better have the right type
+  code_typet call_type = to_code_type(function.type());
+
+  // refine the type in case the forward declaration was incomplete
+  if(call_type.has_ellipsis() &&
+     call_type.parameters().empty())
+  {
+    call_type.remove_ellipsis();
+    forall_expr(it, code.arguments())
```

Comment:
I know you tried to preserve the style of this file, but couldn't we have this as a `for-each` loop?

Reply:
Something like:
```
    for(const auto &argument : code.arguments())
      call_type.parameters().emplace_back(argument.type());
```